systemtest/src/main/java/io/strimzi/systemtest/parallel/SuiteThreadController.java

```diff
@@ -103,8 +119,68 @@ public synchronized void waitUntilEntryIsOpen(ExtensionContext extensionContext)
         LOGGER.debug("{} has locked the @IsolatedSuite and other @IsolatedSuites must wait until lock is released.", extensionContext.getRequiredTestClass().getSimpleName());
     }
 
+    public boolean isRunningAllowedNumberTestSuitesInParallel() {
+        return runningTestSuitesInParallelCount.get() <= maxTestSuitesInParallel;
+    }
+
+    /**
+     * Synchronise point where {@link io.strimzi.systemtest.annotations.ParallelSuite} end it in situation when
+     * Junit5 {@link java.util.concurrent.ForkJoinPool} spawn additional threads, which can exceed limit specified
+     * and thus many threads can start execute test suites which could potentially destroy cluster. This is mechanism,
+     * which will all additional threads (i.e., not needed) put into waiting room. After one of the
+     * {@link io.strimzi.systemtest.annotations.ParallelSuite} is done with execution we release
+     * ({@link #notifyParallelSuiteToAllowExecution(ExtensionContext)} one test suite setting {@code isParallelSuiteReleased} flag.
+     * This ensures that only one test suite will continue with execution and others will still wait.
+     *
+     * @param extensionContext Extension context for test suite name
+     */
+    public void waitUntilAllowedNumberTestSuitesInParallel(ExtensionContext extensionContext) {
+        final String testSuiteToWait = extensionContext.getRequiredTestClass().getSimpleName();
+        waitingTestSuites.add(testSuiteToWait);
+
+        while (!isRunningAllowedNumberTestSuitesInParallel()) {
+            LOGGER.debug("{} is waiting to proceed with execution but current thread exceed maximum " +
```

Comment:
```suggestion
            LOGGER.trace("{} is waiting to proceed with execution but current thread exceed maximum " +
```

And put some debug log above while would be nice